Add Live Channels Demo

RikHeijdens opened the pull request:
Added a demo that demonstrates how the JW Live V2 API can be leveraged to replace a VOD asset that plays on a page in a loop with a livestream, as soon as the livestreaming Channel becomes active.

kcorneli201 commented on demos/live-channels/demo.js:
```diff
@@ -0,0 +1,156 @@
+/** The player config to use in order to initialize the player */
+const VOD_CONFIG = {
+  playlist: 'https://cdn.jwplayer.com/v2/media/8L4m9FJB',
+  // Repeat the VOD indefinitely while we wait for the livestream to become available.
```
// Repeat the VOD indefinitely while we wait for the live event stream to become available.

kcorneli201 commented on demos/live-channels/demo.js:
```diff
@@ -0,0 +1,156 @@
+});
+
+/**
+ * Periodically checks whether the specified livestream channel is available, and if it is, configures the player
```
Periodically checks whether the specified live event stream is available...etc.

RikHeijdens replied:
We actually check the status of the channel here, not the event.

kcorneli201 commented on demos/live-channels/demo.js:
```diff
@@ -0,0 +1,156 @@
+  getChannelStatus(channelId).then((channelStatus) => {
+    console.log(`Received channel status: %O`, channelStatus);
+    if (channelStatus['status'] === 'active') {
+      // Determine the id of the active event based on the returned status.
```
// Determine the ID....etc.

kcorneli201 commented on demos/live-channels/demo.js:
```diff
@@ -0,0 +1,156 @@
+    if (channelStatus['status'] === 'active') {
+      // Determine the id of the active event based on the returned status.
+      const eventId = channelStatus['current_event'];
+      // Attempt to configure the player in order to start livestream playback.
```
// Attempt to configure the player in order to start live event stream playback.

kcorneli201 commented on demos/live-channels/demo.js:
```diff
@@ -0,0 +1,156 @@
+}
+
+/**
+ * (Re-)configures the active playerInstance to play the livestream identified by eventId.
```
* (Re-)configures the active playerInstance to play the live event stream identified by eventId.

kcorneli201 commented on demos/live-channels/demo.js:
```diff
@@ -0,0 +1,156 @@
+ * (Re-)configures the active playerInstance to play the livestream identified by eventId.
+ */
+async function configurePlayer(eventId) {
+  // There may be a slight delay between the livestream becoming available, and its playlist to become available.
```
// There may be a slight delay between the live event stream becoming available, and its playlist to become available.

kcorneli201 commented on demos/live-channels/demo.js:
```diff
@@ -0,0 +1,156 @@
+async function configurePlayer(eventId) {
+  // There may be a slight delay between the livestream becoming available, and its playlist to become available.
+  // Therefore, we first attempt to fetch the playlist for the new live event, as soon as we have successfully fetched
+  // a playlist, we will load it on the player and start playback of the livestream.
```
// a playlist, we will load it on the player and start playback of the live event stream.

kcorneli201 commented on demos/live-channels/demo.js:
```diff
@@ -0,0 +1,156 @@
+}
+
+/**
+ * Fetches the current status of a JW Live 2.0 channel.
```
* Fetches the current status of a live event stream within a Live Channel.

kcorneli201 commented on demos/live-channels/demo.js:
```diff
@@ -0,0 +1,156 @@
+}
+
+/**
+ * Fetches a JW Platform playlist for a particular media item.
```
* Fetches a playlist for a particular media item.

RikHeijdens replied:
I have intentionally used the JW Platform playlist wording here in order to disambiguate between HTTP Live Streaming playlists and JW Platform playlists.

kcorneli201 commented on demos/live-channels/index.html:
```diff
@@ -0,0 +1,32 @@
+<div id="player"></div>
+
+<form id="demo-form" autocomplete="off">
+	<input class="ds-textinput" type="text" id="channel-input" placeholder="Enter a channel id" />
```
Enter a channel ID

kcorneli201 commented on demos/live-channels/index.html:
```diff
@@ -0,0 +1,32 @@
+</form>
+
+<p>
+	Utilize live streaming with JW Platform and JW Player by using JW Live as
+	your live streaming provider.
```
Utilize JW Player's Live Channels as your live event streaming solution.

kcorneli201 commented on demos/live-channels/index.html:
```diff
@@ -0,0 +1,32 @@
+</p>
+<p>
+	This demo demonstrates how to place a player on your page which will play a
+	VOD asset in a loop until a JW Live Channel becomes available.
```
VOD asset in a loop until a live event stream becomes available.

kcorneli201 commented on demos/live-channels/index.html:
```diff
@@ -0,0 +1,32 @@
+</ol>
+<p>
+	After hitting "Select Channel" the code on this page will automatically check for availability
+	of the channel in the background and replace the VOD asset in the player with the livestream
```
of a live event stream in the Live Channel in the background and replace the VOD asset in the player with the live event stream

kcorneli201 commented on demos/live-channels/index.html:
```diff
@@ -0,0 +1,32 @@
+</p>
+<p>
+	<a href="https://support.jwplayer.com/articles/how-to-setup-dvr-live-streaming">Visit our support site</a>
+	for a tutorial on live streaming with JW Platform and JW Player, or
```
for a tutorial on streaming a live event with JW Player, or

kcorneli201 commented on demos/live-channels/index.html:
```diff
@@ -0,0 +1,32 @@
+	<a href="https://support.jwplayer.com/articles/how-to-setup-dvr-live-streaming">Visit our support site</a>
+	for a tutorial on live streaming with JW Platform and JW Player, or
+	<a href="https://www.jwplayer.com/contact-us/?utm_source=developer&utm_medium=CTA&utm_campaign=player-demos">contact us</a>
+	for more information on JW Live Channels.
```
for more information on Live Channels.

RikHeijdens commented:
Thanks for reviewing @kcorneli201, I've addressed most of your comments in order to ensure that our language is consistent between our documentation and demos.

I also left some occurrences of the term `livestream` because some of these comments apply to HTTP Live Streams (as in [RFC-8216](https://tools.ietf.org/html/rfc8216)) in general and not necessarily to JW Player Live Streams.
